src/registrar/views/utility/mixins.py

```diff
@@ -154,6 +154,28 @@ def has_permission(self):
         return True
 
 
+class DomainApplicationPermissionWithdraw(PermissionsLoginMixin):
+
+    """Does the logged-in user have access to withdraw this domain application?"""
+
+    def has_permission(self):
+        """Check if this user has access to withdraw this domain application."""
+        if not self.request.user.is_authenticated:
+            return False
+
+        # user needs to be the creator of the application
+        # this query is empty if there isn't a domain application with this
+        # id and this user as creator
+        if not DomainApplication.objects.filter(creator=self.request.user, id=self.kwargs["pk"]).exists():
+            return False
```

Comment:
(Q) Does this block an admin user (full_access_group) from withdrawing applications if they are not the creator?

Reply:
I believe so! There's another test called `test_application_status_no_permissions` that tests that users can't access or withdraw applications they didn't create

Reply:
I'd double check if that is intended (even as I look through tests I'm second guessing myself LOL), as I feel maybe admin user (full_access_group) should still have power to withdraw even if restricted just in case?